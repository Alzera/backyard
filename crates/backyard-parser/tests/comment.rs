use backyard_parser::parse;

#[test]
fn line_basic() {
  let asts = parse(true, "// test").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn line_long() {
  let asts = parse(true, "///////////// TEST /////////////").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn block() {
  let asts = parse(true, "/*\ntest\n*/").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn doc() {
  let asts = parse(true, "/**\n * test\n */").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn before_block() {
  let asts = parse(true, "if (false) // test
{
}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn after_block() {
  let asts = parse(true, "do {
}
// test
while(false);").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn at_block_end() {
  let asts = parse(true, "if (false) {
}
// test").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn at_statement_end() {
  let asts = parse(true, "$a = 5 /* test */;").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
