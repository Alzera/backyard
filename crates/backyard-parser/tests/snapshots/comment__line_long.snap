---
source: crates/backyard-parser/tests/comment.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: comment_line
    comment:
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 32
      - 84
      - 69
      - 83
      - 84
      - 32
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
      - 47
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 1
        column: 0
        offset: 0
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 0
    offset: 0
leadings: ~
trailings: ~
